src/runtime/objmodel.cpp

```diff
@@ -1705,8 +1711,37 @@ void setattrInternal(Box* obj, const std::string& attr, Box* val, SetattrRewrite
         // We don't need to to the invalidation stuff in this case.
         return;
     } else {
+        // Finally, check __setattr__
+        if (obj->cls->tp_setattr) {
+            rewrite_args = NULL;
+            REWRITE_ABORTED("");
+
+            int rtn = obj->cls->tp_setattr(obj, const_cast<char*>(attr.c_str()), val);
```

Comment:
We've been changing the C API headers to use `const char*`s rather than casting away the constness -- maybe we could do that for tp_setattr?  I guess that would require changing a number of extension modules though.